From 3d447d15299c7eab3058e259c2514d4b407d8de1 Mon Sep 17 00:00:00 2001
From: Joshua Riek <jjriek@verizon.net>
Date: Wed, 7 Aug 2024 10:19:47 -0400
Subject: [PATCH 05/15] arm64: dts: rockchip: Split pcie30x1m1 pinctrl

The PCIe 3.0 PHYs need an external clock and will assert CLKREQ# to
get it. Some RK3588 boards such as the Turning RK1, Mixtile 3588E,
and the ArmSoM AIM7 only provide this clock when CLKREQ# is asserted.

Signed-off-by: Joshua Riek <jjriek@verizon.net>
---
 .../boot/dts/rockchip/rk3588-base-pinctrl.dtsi   | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-base-pinctrl.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-base-pinctrl.dtsi
index ca006f5a4c90..6de5729d2e65 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-base-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-base-pinctrl.dtsi
@@ -1799,12 +1799,22 @@ pcie30x4m0_pins: pcie30x4m0-pins {
 		};
 
 		/omit-if-no-ref/
-		pcie30x4m1_pins: pcie30x4m1-pins {
+		pcie30x4_clkreqn_m1: pcie30x4-clkreqn-m1 {
 			rockchip,pins =
 				/* pcie30x4_clkreqn_m1 */
-				<4 RK_PB4 4 &pcfg_pull_none>,
+				<4 RK_PB4 4 &pcfg_pull_down>;
+		};
+
+		/omit-if-no-ref/
+		pcie30x4_perstn_m1: pcie30x4-perstn-m1 {
+			rockchip,pins =
 				/* pcie30x4_perstn_m1 */
-				<4 RK_PB6 4 &pcfg_pull_none>,
+				<4 RK_PB6 4 &pcfg_pull_none>;
+		};
+
+		/omit-if-no-ref/
+		pcie30x4_waken_m1: pcie30x4-waken-m1 {
+			rockchip,pins =
 				/* pcie30x4_waken_m1 */
 				<4 RK_PB5 4 &pcfg_pull_none>;
 		};
-- 
2.25.1

